assignments/introductory-assignments.md

```diff
@@ -488,14 +582,767 @@ A digital circuit computer with 2GB of memory can distinguish 2^{17 179 869 184}
+
+### Two's complement
+
+Integers are also commonly represented by using [two's complement](https://en.wikipedia.org/wiki/Two%27s_complement) notation, in which the left-most bit indicates the sign: If the leading bit is 1, we subtract 2^{n} to get the [integer](https://en.wikipedia.org/wiki/Integer_(computer_science)) corresponding to an n-bit number in this notation. Two's complement is used to change subtraction to addition by using the "complementary" of a number.
+
+**Using the binary number system**:
+
+Unsigned:
+
+- an unsigned byte can express the numbers 0 .. 255
+- two unsigned bytes can express the numbers 0 .. 65535
+
+[Signed](https://en.wikipedia.org/wiki/Signed_number_representations) (two's complement):
+
+- a signed byte can express the numbers -128 .. 127
+- two signed bytes can express the numbers -32768 .. 32767
+
+
+For example, -1 is the signed byte 0xff or 0b11111111. In this way, a signed byte can express the numbers -128 .. 127. That means we can use the formula -2^{n-1} to 2^{n-1}-1 to calculate the range.
+
+#### From decimal digit into two's complement
+
+If we want to show a positive representation, we can show this just as in the one's complement, so the sign bit is zero.
+
+If we want to represent 41 in two's complement, we fill in the needed bits.
+
+| bit values |  +/-  |  64   |  32   |  16   |   8   |   4   |   2   |   1   | Decimal Value |
+| :--------- | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | ------------: |
+| bits       |   0   |   0   |   1   |   0   |   1   |   0   |   0   |   1   |            41 |
+
+
+To show the value of -41 in two's complement, we need to follow a **three-step plan**.
+
+**First**, we convert the decimal digit into the positive value in binary.
+
+| bit values |  +/-  |  64   |  32   |  16   |   8   |   4   |   2   |   1   | Decimal Value |
+| :--------- | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | ------------: |
+| bits       |   0   |   0   |   1   |   0   |   1   |   0   |   0   |   1   |            41 |
+
+
+**Second**, we invert all binary digits, so that a 0 converts into a 1 and a 1 converts into a 0. 
+
+In the example with -41 that looks as follows:
+
+| bit values |  +/-  |  64   |  32   |  16   |   8   |   4   |   2   |   1   | Decimal Value |
+| :--------- | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | ------------: |
+| bits       |   1   |   1   |   0   |   1   |   0   |   1   |   1   |   0   |           -86 |
+
+We got the `MSB'with 1 as sign bit.
+
+As **third step**, we need to add 1 to the result of step two.
+
+| bit values |  +/-  |  64   |  32   |  16   |   8   |   4   |   2   |   1   | Decimal Value |
+| :--------- | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | ------------: |
+| bits       |   1   |   1   |   0   |   1   |   0   |   1   |   1   |   1   |           -87 |
+
+That means the value of -87 is the two's complement representation of -41.
```

Comment:
When we add -87 + (- 41) = -128, that is the smallest representation of 8 bits.
Could be easier to understand.